packages/core-react/src/hooks/useConnectorInfo.ts

```diff
@@ -0,0 +1,44 @@
+import { useAccount } from 'wagmi';
+import { SafeConnector } from 'wagmi/connectors/safe';
+import {
+  LedgerHIDConnector,
+  LedgerLiveConnector,
+} from '@reef-knot/ledger-connector';
+import { isMobileOrTablet } from '../helpers/userAgents';
+
+type ConnectorInfo = {
+  connectorName?: string;
+  isGnosis: boolean;
+  isLedger: boolean;
+  isLedgerLive: boolean;
+  isDappBrowser: boolean;
+  isAutoConnectionSuitable: boolean;
+};
+
+export const useConnectorInfo = (): ConnectorInfo => {
+  const { connector } = useAccount();
+
+  const isLedger = connector instanceof LedgerHIDConnector;
+  const isLedgerLive = connector instanceof LedgerLiveConnector;
+  const isGnosis = connector instanceof SafeConnector;
+  const isDappBrowser = !!globalThis.window?.ethereum && isMobileOrTablet;
+  const isAutoConnectionSuitable = isLedgerLive || isGnosis || isDappBrowser;
+
+  const connectorName = (() => {
```

Comment:
Not sure what is the purpose of self-calling func here?

Reply:
For historical reasons, previously there were a lot of checks inside this function and it was convenient to write it like this. But now it should be rewritten, yes.